src/dosext/mfs/mfs.c

```diff
@@ -1680,6 +1681,56 @@ dos_fs_dev(struct vm86_regs *state)
     return TRUE;
   }
 
+  /* Redirect all fatfs disks */
+  if (WORD(state->ebx) == DOS_SUBHELPER_MFS_REDIR_MAP_ALL_DRV) {
+#if DOSEMU
+    int i, j, ret=0;
+    char s[255];
+
+    Debug0((dbg_fd, "Redirect fatfs drives\n"));
+
+    for (i = 0; i < MAX_HDISKS; i++) {
+      if (hdisktab[i].type == DIR_TYPE && hdisktab[i].fatfs) {
+        snprintf(s, sizeof(s), LINUX_RESOURCE "%s", hdisktab[i].dev_name);
+        j = RedirectDisk(i + 2, s, hdisktab[i].rdonly);
+        Debug0((dbg_fd, "Redirect %c: %s (err = %d)\n", 'C' + i, j ? "failed" : "ok", j));
+
+        ret += j;
+      }
+    }
+
+    SETWORD(&(state->eax), ret == 0);
+    return TRUE;
+#else
+    return FALSE;
+#endif
+  }
+
+  /* Redirect all printers */
+  if (WORD(state->ebx) == DOS_SUBHELPER_MFS_REDIR_MAP_ALL_PRN) {
```

Comment:
OTOH, if it will start to enumerate them
and the API as you proposed now, then
that API will have to be changed to the
small init and enumerators.